tests/unit/fs_mock_metadata_to_disk_buffer_test.cc

```diff
@@ -0,0 +1,398 @@
+    }
+}
+
+// the following test checks if multiple actions data are correctly added to actions vector
+SEASTAR_THREAD_TEST_CASE(actions_index_test) {
+    mock_metadata_to_disk_buffer buf(default_buff_size, default_alignment);
+    buf.init(0);
+    auto dev_impl = make_shared<mock_block_device_impl>();
+    block_device dev(dev_impl);
+
+    BOOST_REQUIRE_EQUAL(buf.append(ondisk_next_metadata_cluster {1}), APPENDED);
+    buf.flush_to_disk(dev).get();
+    BOOST_REQUIRE_EQUAL(buf.append(ondisk_create_inode {4, 1, {5, 2, 6, 8, 4}}), APPENDED);
+    BOOST_REQUIRE_EQUAL(buf.append(ondisk_update_metadata {4, {5, 2, 6, 8, 4}}), APPENDED);
+    BOOST_REQUIRE_EQUAL(buf.append(ondisk_delete_inode {1}), APPENDED);
+    BOOST_REQUIRE_EQUAL(buf.append(ondisk_medium_write {1, 8, 4, 6, 9}), APPENDED);
+    buf.flush_to_disk(dev).get();
+    BOOST_REQUIRE_EQUAL(buf.append(ondisk_large_write {6, 8, 2, 5}), APPENDED);
+    BOOST_REQUIRE_EQUAL(buf.append(ondisk_large_write_without_mtime {6, 8, 1}), APPENDED);
+    buf.flush_to_disk(dev).get();
+    BOOST_REQUIRE_EQUAL(buf.append(ondisk_truncate {64, 28, 62}), APPENDED);
+    BOOST_REQUIRE_EQUAL(buf.append(ondisk_mtime_update {4, 26}), APPENDED);
+    buf.flush_to_disk(dev).get();
+
+    BOOST_CHECK(buf.is_append_type<ondisk_next_metadata_cluster>(0));
+    BOOST_CHECK(buf.is_type<flush_to_disk>(1));
+    BOOST_CHECK(buf.is_append_type<ondisk_create_inode>(2));
+    BOOST_CHECK(buf.is_append_type<ondisk_update_metadata>(3));
+    BOOST_CHECK(buf.is_append_type<ondisk_delete_inode>(4));
+    BOOST_CHECK(buf.is_append_type<ondisk_medium_write>(5));
+    BOOST_CHECK(buf.is_type<flush_to_disk>(6));
+    BOOST_CHECK(buf.is_append_type<ondisk_large_write>(7));
+    BOOST_CHECK(buf.is_append_type<ondisk_large_write_without_mtime>(8));
+    BOOST_CHECK(buf.is_type<flush_to_disk>(9));
+    BOOST_CHECK(buf.is_append_type<ondisk_truncate>(10));
+    BOOST_CHECK(buf.is_append_type<ondisk_mtime_update>(11));
+    BOOST_CHECK(buf.is_type<flush_to_disk>(12));
+}
+
+// the folowing test checks that constructed buffers are distinct and correctly added to created_buffers vector
+SEASTAR_THREAD_TEST_CASE(virtual_constructor_test) {
+    auto &created_buffers = mock_metadata_to_disk_buffer::created_buffers;
+
+    mock_metadata_to_disk_buffer buf(default_buff_size, default_alignment);
+    buf.init(0);
+    BOOST_REQUIRE_EQUAL(buf.bytes_left(), default_buff_size - checkpoint_size);
+    BOOST_REQUIRE_EQUAL(buf.append(ondisk_delete_inode {1}), APPENDED);
+
+    auto buf2 = buf.virtual_constructor(default_buff_size, default_alignment);
+    buf2->init(default_alignment);
+
+    BOOST_REQUIRE_EQUAL(created_buffers.size(), 1);
+    BOOST_CHECK(created_buffers[0] == buf2);
+    BOOST_REQUIRE_EQUAL(created_buffers[0]->actions.size(), 0);
+    buf2->append(ondisk_delete_inode {1});
+    BOOST_REQUIRE_EQUAL(created_buffers[0]->actions.size(), 1);
+
+    auto buf3 = buf2->virtual_constructor(default_buff_size, default_alignment);
```

Comment:
Maybe it is a good idea to move it and add buf4 here above buf2->append, so that we check that only buf2 is appended and add the same checks that ale below, but for buf4?